pkg/egressgateway/manager.go

```diff
@@ -185,8 +185,8 @@ func NewEgressGatewayManager(p Params) (out struct {
 }, err error) {
 	dcfg := p.DaemonConfig
 
-	// TODO: deprecate --enable-ipv4-egress-gateway, create new --enable-egress-gateway
-	if !dcfg.EnableIPv4EgressGateway {
+	// Check if either the new flag or the deprecated flag is enabled
```

Comment:
I think deprecating the old and introducing the new flag could also be split into a separate PR.